addons/l10n_latam_check/models/account_payment.py

```diff
@@ -42,8 +52,9 @@ def action_post(self):
         # also, changing partner recompute payment method so all checks would be cleaned
         for payment in self.filtered(lambda x: x.l10n_latam_new_check_ids and not x._is_latam_check_payment(check_subtype='new_check')):
             payment.l10n_latam_new_check_ids.unlink()
-        for payment in self.filtered(lambda x: x.l10n_latam_move_check_ids and not x._is_latam_check_payment(check_subtype='move_check')):
-            payment.l10n_latam_move_check_ids = False
+        if not self.env.context.get('l10n_ar_skip_remove_check'):
```

Comment:
I'm adding this because before the change from william, we used the internal_transfer functionality. When creating a payment with internal transfer it will copy the current payment and modify some of the information and so it wasn't going through this action_post function. 
In the case we do a check transfer with a destination_journal like the bank, the payment_method will be computed and will be "manual" for example. So the _is_latam_check_payment will return false and we will remove the check from the payments which is not what we want when being in a internal transfer.
So in case of a internal transfer i'm adding a context key to no remove the checks

Reply:
@malb-odoo Do we still need this, even with the return checks payment method?

Reply:
I think we do yes :grimacing: